processing/src/main/java/org/apache/carbondata/processing/store/CarbonFactDataHandlerModel.java

```diff
@@ -316,31 +326,19 @@ public static CarbonFactDataHandlerModel getCarbonFactDataHandlerModel(CarbonLoa
       String[] tempStoreLocation, String carbonDataDirectoryPath) {
 
     // for dynamic page size in write step if varchar columns exist
-    List<Integer> varcharDimIdxInNoDict = new ArrayList<>();
```

Comment:
Is the pageSize validation not applicable for compaction flow?

Reply:
It will be used for compaction also, 
getCarbonFactDataHandlerModel() is called in compaction flow, I have handled filling the datatype list in that method.